modin/experimental/engines/omnisci_on_ray/frame/data.py

```diff
@@ -1214,41 +1215,28 @@ def _arrow_col_slice(self, new_columns):
 
     def _arrow_row_slice(self, row_numeric_idx):
         table = self._execute_arrow()
+
+        if not isinstance(row_numeric_idx, slice) and not is_range_like(
+            row_numeric_idx
+        ):
+            if not isinstance(row_numeric_idx, (pyarrow.Array, np.ndarray, list)):
+                row_numeric_idx = pyarrow.array(row_numeric_idx)
+            return table.take(row_numeric_idx)
+
         if isinstance(row_numeric_idx, slice):
-            start = 0 if row_numeric_idx.start is None else row_numeric_idx.start
-            if start < 0:
-                start = table.num_rows - start
-            end = (
-                table.num_rows if row_numeric_idx.stop is None else row_numeric_idx.stop
-            )
-            if end < 0:
-                end = table.num_rows - end
-            if row_numeric_idx.step is None or row_numeric_idx.step == 1:
-                length = 0 if start >= end else end - start
-                return table.slice(start, length)
-            else:
-                parts = []
-                for i in range(start, end, row_numeric_idx.step):
-                    parts.append(table.slice(i, 1))
-                return pyarrow.concat_tables(parts)
-
-        start = None
-        end = None
-        parts = []
-        for idx in row_numeric_idx:
-            if start is None:
-                start = idx
-                end = idx
-            elif idx == end + 1:
-                end = idx
-            else:
-                if start:
-                    parts.append(table.slice(start, end - start + 1))
-                start = idx
-                end = idx
-        parts.append(table.slice(start, end - start + 1))
+            row_numeric_idx = range(*row_numeric_idx.indices(table.num_rows))
```

Comment:
could you explain please what this does?

Reply:
`slice.indices` returns a tuple of start, stop positions, and the step for a sequence of the given length, that tuple is then converted into a range, so we can easily retrieve the slice length further. This construction looks simpler than the previous if-else tree.